utils/rowgroup/rowgroup.h

```diff
@@ -1003,24 +1053,31 @@ inline void Row::setStringField(const utils::ConstString& str, uint32_t colIndex
   }
   else
   {
-    memcpy(&data[offsets[colIndex]], str.str(), length);
-    memset(&data[offsets[colIndex] + length], 0, offsets[colIndex + 1] - (offsets[colIndex] + length));
-  }
-}
+    if (length > getColumnWidth(colIndex))
+      length = getColumnWidth(colIndex);
 
-inline std::string Row::getVarBinaryStringField(uint32_t colIndex) const
-{
-  if (inStringTable(colIndex))
-    return getConstString(colIndex).toString();
-
-  return std::string((char*)&data[offsets[colIndex] + 2], *((uint16_t*)&data[offsets[colIndex]]));
+    uint8_t* buf = &data[offsets[colIndex]];
+    if (str.str())
+    {
+      memcpy(buf, str.str(), length);
+      memset(buf + length, 0, offsets[colIndex + 1] - (offsets[colIndex] + length));
+    }
+    else
+    {
+      setToNull(colIndex);
```

Comment:
I am curious if you tried to remove this setToNull() call b/c it was not here originally.

Reply:
It cannot be removed.

While we usually use rowgroup to fill it once and then only read, nothing in the interface precludes some creative person to set fields several times. Thus, it is possible that the aforementioned creative persong can set a field in a row to a non-NULL value and then to NULL. setToNull lets this to be the a valid use case.

Reply:
Notable exception is group by processing and likes - they read and write the same rowgroup.

This process usually does not set NULL after setting a non-NULL value, but it is safe to assume someone will.

Reply:
It is not called every time here. It is called only when ConstString is NULL (nullptr). Which is not all that often.

Also, there were not setToNull there because empty string was a NULL indicator.

Thus, I very much like setToNull() to be there until it is proven to be a detriment performance-wise.

Reply:
But!

I can make a specialized fast version for setToNull, which will just set the flag. I'll do that.

Reply:
Actually, it was superfluous because of setNullMark few lines above. Deleted that.

Reply:
The setToNull() call is essential for handling short strings.

Reply:
setToNull is needed - it sets magic values for short character columns. I can speed up code by partially inlining it here but it will produce more copy-paste.

Right now I am on the "make it right" side of development on what is going on here.